### PR TITLE
Fix ListRepositories and CreateBuild

### PR DESCRIPTION
**Pull request by ChadNedzlek**
These 2 methods had invalid EntityFramework calls that broke in the migration to 3.1

Also added tests for them, which necessitated installing (and managing) SQL Express
on the build machine.  It only takes around 4 minutes to install, so that's ok,
and devs will already have it installed so they are fast to run locally.

### CODE REVIEW

File: eng/Install-SqlExpress.ps1 [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+$packageName= 'sql-server-express'
+$url        = 'https://download.microsoft.com/download/8/4/c/84c6c430-e0f5-476d-bf43-eaaa222a72e0/SQLEXPR_x64_ENU.exe'
+$silentArgs = "/IACCEPTSQLSERVERLICENSETERMS /Q /ACTION=install /INSTANCEID=SQLEXPRESS /INSTANCENAME=SQLEXPRESS /UPDATEENABLED=FALSE"
+
+$tempDir = Join-Path (Get-Item $env:TEMP).FullName "$packageName"
+
+if (![System.IO.Directory]::Exists($tempDir)) { [System.IO.Directory]::CreateDirectory($tempDir) | Out-Null }
+$fileFullPath = "$tempDir\SQLEXPR.exe"
+
+(New-Object System.Net.WebClient).DownloadFile($url, $fileFullPath )
```
Comment (missymessa) [inline eng/Install-SqlExpress.ps1:13]: Do we want to do something if the url for the download changes?

**Comment by missymessa** [comment]
If a dev were to run these tests locally and didn't have SQLExpress set up, are there error messages telling them that they need to set that up? (Even though https://github.com/dotnet/arcade-services/blob/master/src/Maestro/Readme.md calls out these set up steps)

File: eng/Install-SqlExpress.ps1 (hunk 1 above)
Comment (ChadNedzlek) [reply eng/Install-SqlExpress.ps1:13]: The download.microsoft.com/download URL are very shelf stable.  As long as the product exists, so will this URL.

(It's also the URL the chocolatey installer uses)

Comment (ChadNedzlek) [reply eng/Install-SqlExpress.ps1:13]: This will also throw some exception if it happens. :-)

**Comment by ChadNedzlek** [comment]
The error message will be that it can't connect to the "SQLEXPRESS" database, so that should be pretty clear.

**Pull request merged**